src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1650,11 +1650,6 @@ internal static CollectionExpressionTypeKind GetCollectionExpressionTypeKind(CSh
             {
                 return CollectionExpressionTypeKind.CollectionBuilder;
             }
-            else if (implementsSpecialInterface(compilation, destination, SpecialType.System_Collections_Generic_IEnumerable_T))
-            {
-                elementType = default;
-                return CollectionExpressionTypeKind.ImplementsIEnumerableT;
-            }
```

Comment:
Discussed offline:
It feels strange for `GetCollectionExpressionTypeKind` to introduce a concept of "element type" in some cases.
Could we align on the concept of iteration type for all kinds of collections and add the special cases to `TryGetCollectionIterationType` instead, or call `TryGetCollectionIterationType` here and rename `elementType`->`iterationType`? 
That would simplify concepts and align with the spec better.

Reply:
Thanks for the suggestion. I agree, we should refactor these two methods, either to move calculation of iteration type to `TryGetCollectionIterationType` exclusively, or to have one method call the other. Let's look at refactoring after completing other expected changes here.